iroh/src/node.rs

```diff
@@ -257,16 +295,24 @@ where
     /// This will create the underlying network server and spawn a tokio task accepting
     /// connections.  The returned [`Node`] can be used to control the task as well as
     /// get information about it.
-    pub async fn spawn(self) -> Result<Node<D>> {
+    pub async fn spawn(self) -> Result<Node<D, S>> {
         trace!("spawning node");
         let rt = self.rt.context("runtime not set")?;
 
+        // TODO: this should actually run globally only once.
```

Comment:
we should do sth about this before merging

Reply:
Actually I think we can just remove the TODO:
`init_metrics_collection` calls `iroh_metrics::core::Core::try_init(|reg, metrics| { ... }).ok()`, whereas `Core::try_init` tries to set a `OnceCell`, which will fail for subsequent invocations, which is silenced by the `ok`.
So - not sure how/if we want to do this differently?

Reply:
The only worry I have with this setup, is that we now can not add any metrics in `main.rs` eg, because the first try_init will succeed and follow ups will not, so only the metrics in here will be added. It is probably be fine but want to call it out.

cc @Arqu

Reply:
Well, you would have to add the metrics in `main.rs` before spawning the node. This should be easily doable.
I added a commit that clarifies the current setup with better method names and comments: https://github.com/n0-computer/iroh/pull/1333/commits/503ff23b6439a4787bd4d02e7d605c65cbfd8022
Or do you have another idea on how to proceed with this?

Reply:
I think comments about this are good enough for now 👍

Reply:
Yeah I think having comments is the best option currently. Due to the metrics setup we can only define them once depending where your entrypoint is. The benefit is that you can randomly drop metrics wherever you like without having to pass weirdo references everywhere.